### PR TITLE
Simplify intrinsics of broadcasts to broadcasts of intrinsics

### PR DESCRIPTION
**Pull request by dsharletg**
Currently, things like shift_right(x4(8), x4(1)) are not constant folded. This PR fixes that.

### CODE REVIEW

File: src/Simplify_Call.cpp [hunk 1]
```diff
@@ -95,6 +113,11 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
                op->is_intrinsic(Call::count_trailing_zeros)) {
         Expr a = mutate(op->args[0], nullptr);
 
+        Expr scalar = lift_elementwise_broadcasts(op->type, op->name, {a}, op->call_type);
```
Comment (abadams) [inline src/Simplify_Call.cpp:116]: "scalar" seems like a misnomer here

File: src/Simplify_Call.cpp [hunk 2]
```diff
@@ -74,6 +74,24 @@ int ctz64(uint64_t x) {
...
 
+// Rewrite name(broadcast(args)) to broadcast(name(args)).
+// Assumes that scalars are implicitly broadcasted.
+Expr lift_elementwise_broadcasts(Type type, const std::string &name, std::vector<Expr> args, Call::CallType call_type) {
+    if (type.lanes() == 1) {
+        return Expr();
+    }
+    for (Expr &i : args) {
+        const Broadcast *b = i.as<Broadcast>();
+        if (b) {
```
Comment (abadams) [inline src/Simplify_Call.cpp:85]: nit: `if (const Broadcast *b = i.as<Broadcast>())` would be terser and idiomatic.

**Review by abadams — APPROVED** [review]

File: src/Simplify_Call.cpp [hunk 3]
```diff
@@ -74,6 +74,34 @@ int ctz64(uint64_t x) {
 #endif
 }
 
+// Rewrite name(broadcast(args)) to broadcast(name(args)).
+// Assumes that scalars are implicitly broadcasted.
```
Comment (steven-johnson) [inline src/Simplify_Call.cpp:78]: super-duper-nit: past tense of 'broadcast' is also 'broadcast'

**Review by steven-johnson — APPROVED** [review]

**Comment by dsharletg** [comment]
The timeouts look unrelated, merging.

**Pull request merged**